src/ExtensionConfigHandler.php

```diff
@@ -352,6 +352,156 @@ protected function getSourceStorageWrapper(array $source_dirs, $subdir = Install
     return $source_storage;
   }
 
+  public function sortAndFilterOutput($dependencies) {
+    // Sort and filter dependency lists, and finally change to simple indexed
+    // array lists. This means the output can then be directly copied for use
+    // in a .info.yml file if the output format is YAML.
+    foreach (array_keys($dependencies) as $dependency_type) {
+      if ($dependencies[$dependency_type]) {
+        sort($dependencies[$dependency_type]);
+        $dependencies[$dependency_type] = array_values($dependencies[$dependency_type]);
+      }
+      else {
+        unset($dependencies[$dependency_type]);
+      }
+    }
+    return $dependencies;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function getConfigDependancies($extension, $limit = NULL) {
+    $dependencies = array();
+    $listed_config = $this->getExtensionInfo($extension, 'managed');
+    foreach ($listed_config as $key) {
+      $config_dependencies = $this->getIndividualConfigDependancies($key, $limit);
+      foreach ($config_dependencies as $dependency_type => $type_dependencies) {
+        if (!isset($dependencies[$dependency_type])) {
+          $dependencies[$dependency_type] = array();
+        }
+        $dependencies[$dependency_type] += $type_dependencies;
+      }
+    }
+    // Exclude 'core' and the specified extension from the full list.
+    unset($dependencies['module']['core']);
+    unset($dependencies['module'][$extension]);
+    return $this->sortAndFilterOutput($dependencies);
+  }
+
+  /**
+   * @param $name
+   *   The config key to find the dependencies of.
+   * @param null $recursion_limit
+   *   Optionally limit the levels of recursion.
+   * @return mixed
+   */
+  public function getIndividualConfigDependancies($name, $recursion_limit = NULL) {
```

Comment:
I left it public because I thought it might be helpful to have accessible. What's achieved by protecting it?

On the other hand I didn't think it necessary to require it of *any* implementor of our interface, so had left it out. But if you'd like things either protected or in the interface, I'll go for the interface.